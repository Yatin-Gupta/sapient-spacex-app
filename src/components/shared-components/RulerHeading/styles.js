/**
 * RulerHeading Stylesheet in JSS
 */
const headingUnderlineColor = "#c1bdbc";
const styles = {
  headingDiv: {
    display: "flex",
    justifyContent: "center",
    width: "100%",
    "&>div": {
      width: "auto",
      padding: "0px 5px 2px 5px",
      border: "0px",
      textAlign: "center",
      borderBottom: `2px solid ${headingUnderlineColor}`,
      "& span": {
        letterSpacing: "1px",
      },
    },
  },
};

export default styles;
