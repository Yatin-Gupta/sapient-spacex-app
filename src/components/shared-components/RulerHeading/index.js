import React from "react";
import { createUseStyles } from "react-jss";
import styles from "./styles";

const useStyles = createUseStyles(styles);

/**
 * Ruler Heading Component
 * @param {*} props
 */
function RulerHeading(props) {
  const classes = useStyles();
  let className = props.className ? `${classes.headingDiv} ${props.className}` : classes.headingDiv;

  return (
    <React.Fragment>
      <div {...props} className={className}>
        <div role="heading" aria-level="2">
          <span>{props.text}</span>
        </div>
      </div>
    </React.Fragment>
  );
}

export default RulerHeading;
