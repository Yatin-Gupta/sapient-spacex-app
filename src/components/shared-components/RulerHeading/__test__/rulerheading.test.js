import React from "react";
import ReactDOM from "react-dom";
import RulerHeading from "../index";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("RulerHeading component renders without crashing", () => {
  const div = document.createElement("div");
  const testHeading = "Test Cases";
  ReactDOM.render(<RulerHeading text={testHeading} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("RulerHeading component renders with text attribute passed as props", () => {
  const testHeading = "Test Cases";
  const { getByTestId } = render(<RulerHeading text={testHeading} />);
  expect(getByTestId("heading")).toHaveTextContent(testHeading);
});

it("RulerHeading component matches the snapshot", () => {
  const testHeading = "Test Cases";
  const tree = renderer.create(<RulerHeading text={testHeading} />).toJSON();
  expect(tree).toMatchSnapshot();
});
