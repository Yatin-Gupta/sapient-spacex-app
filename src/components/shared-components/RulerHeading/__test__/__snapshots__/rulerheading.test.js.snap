// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RulerHeading component matches the snapshot 1`] = `
<div
  className="headingDiv-0-2-1"
  text="Test Cases"
>
  <div
    aria-level="2"
    role="heading"
  >
    <span
      data-testid="heading"
    >
      Test Cases
    </span>
  </div>
</div>
`;
